spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -1972,6 +1974,223 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
+
+      withSQLConf(DeltaSQLConf.TABLE_FEATURE_DROP_ENABLED.key -> true.toString) {
+        val command = AlterTableDropFeatureDeltaCommand(
+          DeltaTableV2(spark, deltaLog.dataPath),
+          TestRemovableWriterFeature.name)
+        command.run(spark)
+      }
+
+      // Writer feature is removed from the writer features set.
+      val snapshot = deltaLog.update()
+      assert(snapshot.protocol === emptyProtocolWithTableFeatures)
+      assert(!snapshot.metadata.configuration.contains(TestRemovableWriterFeature.TABLE_PROP_KEY))
+      assertPropertiesAndShowTblProperties(deltaLog)
+    }
+  }
+
+  test("Remove a non-removable feature") {
+    withTempDir { dir =>
+      val deltaLog = DeltaLog.forTable(spark, dir)
+      sql(s"""CREATE TABLE delta.`${dir.getCanonicalPath}` (id bigint) USING delta
+             |TBLPROPERTIES (
+             |delta.feature.${TestWriterMetadataNoAutoUpdateFeature.name} = 'supported'
+             |)""".stripMargin)
+
+      val expectedProtocol = protocolWithWriterFeature(TestWriterMetadataNoAutoUpdateFeature)
+      assert(deltaLog.update().protocol === expectedProtocol)
+
+      withSQLConf(DeltaSQLConf.TABLE_FEATURE_DROP_ENABLED.key -> true.toString) {
+        val command = AlterTableDropFeatureDeltaCommand(
+          DeltaTableV2(spark, deltaLog.dataPath),
+          TestWriterMetadataNoAutoUpdateFeature.name)
+
+        val e = intercept[DeltaTableFeatureException] {
+          command.run(spark)
+        }
+        checkError(
+          exception = e,
+          errorClass = "DELTA_FEATURE_DROP_NONREMOVABLE_FEATURE",
+          parameters = Map("feature" -> TestWriterMetadataNoAutoUpdateFeature.name))
+      }
+    }
+  }
+
+  test("Remove a feature not supported by the client") {
+    withTempDir { dir =>
+      val deltaLog = DeltaLog.forTable(spark, dir)
+      sql(s"CREATE TABLE delta.`${dir.getCanonicalPath}` (id bigint) USING delta")
+
+      assert(
+        deltaLog.update().protocol === Protocol(
+          minReaderVersion = 1,
+          minWriterVersion = 2,
+          readerFeatures = None,
+          writerFeatures = None))
+
+      withSQLConf(DeltaSQLConf.TABLE_FEATURE_DROP_ENABLED.key -> true.toString) {
+        val command = AlterTableDropFeatureDeltaCommand(
+          DeltaTableV2(spark, deltaLog.dataPath),
+          "NonSupportedFeature")
+
+        val e = intercept[DeltaTableFeatureException] {
+          command.run(spark)
+        }
+        checkError(
+          exception = e,
+          errorClass = "DELTA_FEATURE_DROP_UNSUPPORTED_CLIENT_FEATURE",
+          parameters = Map("feature" -> "NonSupportedFeature"))
+      }
+    }
+  }
+
+  test("Remove a feature not present in the protocol") {
+    withTempDir { dir =>
+      val deltaLog = DeltaLog.forTable(spark, dir)
+      sql(s"CREATE TABLE delta.`${dir.getCanonicalPath}` (id bigint) USING delta")
+
+      assert(
+        deltaLog.update().protocol === Protocol(
+          minReaderVersion = 1,
```

Comment:
Also test this where the protocol has table features support but simply doesn't include the feature? And how about a test where the protocol has the table feature _implicitly_ (i.e., legacy protocol versions)?